examples/Client/DistributedLock/Controllers/BindingController.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Dapr;
+using Dapr.Client;
+using DistributedLock.Model;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace DistributedLock.Controllers
+{
+    [ApiController]
+    public class BindingController : ControllerBase
+    {
+        private DaprClient client;
+        private ILogger<BindingController> logger;
+        private string appId;
+
+        public BindingController(DaprClient client, ILogger<BindingController> logger)
+        {
+            this.client = client;
+            this.logger = logger;
+            this.appId = Environment.GetEnvironmentVariable("APP_ID");
+        }
+
+        [HttpPost("cronbinding")]
+        [Obsolete]
+        public async Task<IActionResult> HandleBindingEvent()
+        {
+            logger.LogInformation($"Received binding event on {appId}, scanning for work.");
+
+            var request = new BindingRequest("localstorage", "list");
+            var result = client.InvokeBindingAsync(request);
+
+            var rawData = result.Result.Data.ToArray();
+            var files = JsonSerializer.Deserialize<string[]>(rawData);
+
+            if (files != null)
+            {
+                foreach (var file in files.Select(file => file.Split("/").Last()).OrderBy(file => file))
+                {
+                    await AttemptToProcessFile(file);
+                }
+            }
+            
+            return Ok();
+        }
+
+
+        [Obsolete]
+        private async Task AttemptToProcessFile(string fileName)
+        {
+            // We don't need to lock for reading, so try that first to see if the object still exists before locking.
+            if (await GetFile(fileName) == null)
```

Comment:
This pattern is problematic, it's a TOCTOU https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use

You have to check null twice, or just do the check inside the lock.

Reply:
This is a better sample overall though.

Reply:
Yeah, I did that on purpose to show that once you're inside the lock you're safe.

I do agree though that it's not a good practice, it was forced to demo the lock. If you think it just promotes bad practices I'll go ahead and remove it.

Reply:
Chatted offline, removed it to avoid showing people poor practices :)